backend/src/vehicles/vehicles.service.ts

```diff
@@ -99,7 +95,33 @@ export class VehicleService {
     const { id: organizationId } =
       this.organizationDomainService.getRequestOrganization();
 
-    const updatedVehicle = this.vehiclesRepository.update(
+    const vehicleToEdit = await this.vehiclesRepository.findOne({
+      where: {
+        id: vehicleId,
+        organization: { id: organizationId },
+      },
+    });
+
+    if (vehicleToEdit === null) {
+      throw new Error('VEHICLE_NOT_FOUND');
+    }
+
+    const existsVehicleWithSameLicensePlate =
+      await this.vehiclesRepository.countBy({
+        id: Not(vehicleId),
+        licensePlate: vehicle.licensePlate,
+        organization: { id: organizationId },
+      });
+
+    if (existsVehicleWithSameLicensePlate > 0) {
+      throw new Error('VEHICLE_SAME_LICENSE_PLATE');
+    }
+
+    if (vehicle.vin !== undefined && vehicle.vin?.length !== 17) {
+      throw new Error('VIN_LENGTH_NOT_CORRECT');
+    }
```

Comment:
Why are we doing this validation here and not in the DTO?. Please see [the docs for `class-validator`](https://www.npmjs.com/package/class-validator) for more info on that